sql/core/src/test/scala/org/apache/spark/sql/execution/QueryExecutionSuite.scala

```diff
@@ -244,14 +245,19 @@ class QueryExecutionSuite extends SharedSparkSession {
   }
 
   test("SPARK-35378: Eagerly execute non-root Command") {
-    def qe(logicalPlan: LogicalPlan): QueryExecution = new QueryExecution(spark, logicalPlan)
+    val mockCallback = MockCallbackEagerCommand()
+    def qe(logicalPlan: LogicalPlan): QueryExecution = new QueryExecution(
+      spark,
+      logicalPlan,
+      new QueryPlanningTracker(mockCallback.callback))
 
     val showTables = ShowTables(UnresolvedNamespace(Seq.empty[String]), None)
     val showTablesQe = qe(showTables)
     assert(showTablesQe.commandExecuted.isInstanceOf[CommandResult])
```

Comment:
`assert(mockCallback.tracker == null)` before `showTablesQe.commandExecuted`, and `assert(mockCallback.tracker != null)` after